platform/tezos/block.go

```diff
@@ -1,20 +1,117 @@
 package tezos
 
-import "github.com/trustwallet/golibs/types"
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/trustwallet/golibs/coin"
+	"github.com/trustwallet/golibs/types"
+
+	"github.com/itchyny/timefmt-go"
+)
 
 func (p *Platform) CurrentBlockNumber() (int64, error) {
-	return p.client.GetCurrentBlock()
+	return p.rpcClient.GetBlockHead()
 }
 
 func (p *Platform) GetBlockByNumber(num int64) (*types.Block, error) {
-	txTypes := []string{TxTypeTransaction, TxTypeDelegation}
-	srcTxs, err := p.client.GetBlockByNumber(num, txTypes)
+	block, err := p.rpcClient.GetBlockByNumber(num)
 	if err != nil {
 		return nil, err
 	}
-	txs := NormalizeTxs(srcTxs, "")
-	return &types.Block{
-		Number: num,
-		Txs:    txs,
+	return NormalizeRpcBlock(block)
+}
+
+func NormalizeRpcBlock(block RpcBlock) (*types.Block, error) {
+	txs := []types.Tx{}
+
+	for _, ops := range block.Operations {
```

Comment:
Could we simplify it?

Reply:
unfortunately no, if you look at the json response